lib/tower_cli/models/base.py

```diff
@@ -545,6 +545,15 @@ def write(self, pk=None, create_on_missing=False, fail_on_found=False,
         # Determine which record we are writing, if we weren't given a
         # primary key.
         if not pk:
+            # Sanity check: Are we missing required values?
+            required_fields = [i.key or i.name
+                               for i in self.fields if i.required]
+            missing_fields = [i for i in required_fields if i not in kwargs]
+            if missing_fields and not pk:
```

Comment:
Part of the problem here is that `pk` could be defined after the subsequent to `_lookup` is done, and that's part of what was originally being tested for (I think, and I think that's tied up with the test changes you were forced to make). I would have the `missing_fields` calc happen even before the `if not pk:` block, and then maybe use a conditional along with `fail_on_found` to raise another type of error of that happens (this could still be in the original `except` block you used). FOF is a good indicator that this is called by the `create` command.